packages/grafana-prometheus/src/querybuilder/shared/QueryEditorHints.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+
+import { QueryHint } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { Button, Tooltip } from '@grafana/ui';
+
+import { PromQueryEditorProps } from '../../components/types';
+
+export function QueryEditorHints(props: PromQueryEditorProps) {
+  const [hints, setHints] = useState<QueryHint[]>([]);
+
+  useEffect(() => {
+    const query = { expr: props.query.expr, refId: props.query.refId };
+    const hints = props.datasource.getQueryHints(query, props.data?.series || []).filter((hint) => hint.fix?.action);
+    setHints(hints);
+  }, [props.datasource, props.data, props.query]);
+
+  return (
+    <>
+      {hints.length > 0 && (
```

Comment:
if there is no hint can we return early? So we don't need to wrap the whole code block.

Reply:
Sorry, I didn't get you, can you please help me with what are you suggesting?

Reply:
So if there is no hints let's just return null in the beginning. It's not a mandatory change though. Just a preference. 

So it will look like below:

```tsx
 if (hints.length === 0) {
    return null
}

// We have hints 
<div className={styles.container}>
  {hints.map((hint) => {
    return (<hintbuttons/>)
  })}
</div>
```